transport/testing/reqres.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package test_transport
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/yarpc/yarpc-go/transport"
+)
+
+// RequestMatcher may be used in gomock argument lists to assert that two
+// requests match.
+//
+// Requests are considered to be matching if:
+//
+// - All their primitive parameters match.
+// - The headers of the received request include all the headers from the
+//   source request. It may include extra headers.
+// - The contents of the request bodies are the same.
+type RequestMatcher struct {
+	t    *testing.T
+	req  *transport.Request
+	body []byte
+}
+
+// NewRequestMatcher constructs a new RequestMatcher from the given testing.T
+// and request.
+//
+// The request's contents are read in their entirety and replaced with a
+// bytes.Reader.
+func NewRequestMatcher(t *testing.T, r *transport.Request) RequestMatcher {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		t.Fatalf("failed to read request body: %v", err)
+		panic(err.Error())
```

Comment:
why panic as well as `Fatalf`, I think the `Fatalf` will cause the current goroutine to exit anyway